### PR TITLE
Trigraphs

### PR DESCRIPTION
**Pull request by mppf**
In response to some issues brought up by @cassella, fix the parser to
avoid C trigraph pitfalls and to give errors for unsupported C \ escape
sequences.

Before this change, "??!" for example would make it to the C compiler as
a string literal, which would then become "|" (and then Chapel will
compute the wrong length for the string).

This change also introduces errors for octal and unicode escapes, which
are valid for C but not currently valid for Chapel. That way, we won't
accidentally pass them to the C compiler (and get the length of the
string wrong).

Passed full local testing.
Reviewed by @ben-albrecht and @noakesmichael - thanks!

### CODE REVIEW

**Comment by mppf** [comment]
@ben-albrecht - care to review?

File: compiler/include/bison-chapel.h [hunk 1]
```diff
@@ -1,19 +1,19 @@
-/* A Bison parser, made by GNU Bison 3.0.4.  */
+/* A Bison parser, made by GNU Bison 2.5.  */
```
Comment (ben-albrecht) [inline compiler/include/bison-chapel.h]: What's going on here with version numbers of Bison?

**Comment by ben-albrecht** [comment]
@mppf Sure, but I think it would be good to have someone like @noakesmichael take a look as well.

File: compiler/include/bison-chapel.h (hunk 1 above)
Comment (mppf) [inline compiler/include/bison-chapel.h]: As far as I know it doesn't matter... I run 'make parser' on chapcs10, which has 2.5. I think you were the last person before then to update it, and you must have used a newer bison.

Comment (ben-albrecht) [inline compiler/include/bison-chapel.h]: Makes sense.

**Pull request merged**